app/services/bulk_upload/lettings/year2023/row_parser.rb

```diff
@@ -535,13 +543,15 @@ def validate_nulls
       if setup_question?(question)
         fields.each do |field|
           if errors[field].present?
-            errors.add(field, I18n.t("validations.not_answered", question: question.check_answer_label&.downcase), category: :setup)
+            question_text = question.check_answer_label.presence || question.header
+            errors.add(field, I18n.t("validations.not_answered", question: question_text&.downcase), category: :setup)
           end
         end
       else
         fields.each do |field|
           unless errors.any? { |e| fields.include?(e.attribute) }
-            errors.add(field, I18n.t("validations.not_answered", question: question.check_answer_label&.downcase))
+            question_text = question.check_answer_label.presence || question.header
+            errors.add(field, I18n.t("validations.not_answered", question: question_text&.downcase))
```

Comment:
@bibblobcode I've added ` || "this question"` to the end of the line. I think this is fine because we know `question` exists (it was checked earlier in the method) so this just catches the case when the question doesn't have a header or check_answer_label. Let me know what you think! Thanks 🙂

Reply:
Makes sense, thanks!